src/scancode/cli.py

```diff
@@ -217,724 +225,891 @@ def print_version(ctx, param, value):
+                progress_manager = partial(progressmanager,
+                    item_show_func=item_show_func, verbose=verbose, file=sys.stderr)
+
+            # TODO: add CLI option to bypass cache entirely
+            scan_success = scan_codebase(codebase, scanners, processes, timeout,
+                with_timing=timing, with_info=with_info, progress_manager=progress_manager)
+
+            scanned_count, _, scanned_size = codebase.counts(
+                update=True, skip_root=False)
+
+            codebase.summary['scan_names'] = scan_names
+            codebase.summary['scanned_count'] = scanned_count
+            codebase.summary['scanned_size'] = scanned_size
+            codebase.timings['scan'] = time() - scan_start
+
+            success = success and scan_success
+
+        ############################################################################
+        # 7. run postscans
+        ############################################################################
+
+        post_scan_start = time()
+        # TODO: add progress indicator
+        if not quiet and not verbose and post_scan_plugins:
+            echo_stderr('Run post-scan plugins...', fg='green')
+        for name, plugin in post_scan_plugins.items():
+            if not quiet and verbose:
+                echo_stderr('Run post-scan plugin: %(name)s...' % locals(), fg='green')
+
+            try:
+                plugin.process_codebase(codebase)
+            except:
+                msg = 'ERROR: failed to run post-scan plugin: %(name)s:' % locals()
+                echo_stderr(msg, fg='red')
+                echo_stderr(traceback.format_exc())
+                ctx.exit(2)
+            codebase.update_counts()
+
+        codebase.timings['post-scan'] = time() - post_scan_start
+
+        ############################################################################
+        # 8. apply output filters
+        ############################################################################
+        output_filter_start = time()
+        # TODO: add progress indicator
+        if not quiet and not verbose and output_filter_plugins:
+            echo_stderr('Run output filter plugins...', fg='green')
+
+        filters = tuple(plugin.process_resource for plugin in output_filter_plugins.values())
+        if filters:
+            # This is a set of resource ids to filter out from the final outputs
+            filtered_rids_add = codebase.filtered_rids.add
+            try:
+                for rid, resource in codebase.get_resources_with_rid():
+                    if all(to_keep(resource) for to_keep in filters):
+                        continue
+                    filtered_rids_add(rid)
+            finally:
+                msg = 'ERROR: failed to run output filter plugins'
+                echo_stderr(msg, fg='red')
+                echo_stderr(traceback.format_exc())
+                ctx.exit(2)
+
+        codebase.timings['output-filter'] = time() - post_scan_start
```

Comment:
This probably should be `output_filter_start` instead of `post_scan_start`.

Reply:
@haikoschol thanks. Indeed. This is fixed in the latest commit: I now use a single function to run a scan "stage" that is used for all stages except scanners.